### DISCUSSION TITLE
Undefined properties in x-data

### DISCUSSION THREAD

**Discussion by personalsoft-fabianobonin** [alpinejs/alpine#1497] [1. Help] [answered]
Why does this code works:

```html
    <div x-data="{ entity: {} }">
      <input type="text"
        :value="entity.field1">
    </div>
```

While this code doesn't?
It throws "Uncaught (in promise) ReferenceError: field1 is not defined"

```html
    <div x-data="{ entity: {} }">
      <input type="text"
        :value="field1">
    </div>
```

Shouldn't they have the same behavior, as the proxy returns the same value for both expressions?

```javascript
console.log(document.querySelector('div')._x_dataStack[0].entity.field1)
undefined

console.log(document.querySelector('div')._x_dataStack[0].field1)
undefined
```

**Comment by ryangjchandler** [comment, marked as answer]
@personalsoft-fabianobonin This behaviour isn't Alpine's doing, it's natural JavaScript behaviour. If you try to access a property that doesn't exist on an object, it returns `undefined`.

The code that you say produces an error is acting differently. It's trying to find a variable in the current scope called `field1` which doesn't exist.

**Reply by personalsoft-fabianobonin** [reply]
'field1' doesn't exists in "entity" scope also, but it doesn't throws an error.

So can I presume all 'root' properties *must* exist in the x-data scope?

I see no problem on that, I'm just checking to make sure my usage patterns are ok.

**Reply by ryangjchandler** [reply]
Yeah, all root properties must exist in the scope. 

`field1` doesn't exist on the `entity` object but JavaScript behaviour says that this won't error out and will return `undefined` unless defined explicitly.
